### PR TITLE
Add new non-variadic versions of _d_newarraym*T

### PR DESCRIPTION
**Pull request by yebblies**
The old versions can be deleted once the compiler has been updated

### CODE REVIEW

**Comment by yebblies** [comment]
Once this is merged the compiler can be updated to use them: https://github.com/D-Programming-Language/dmd/pull/4303

**Comment by yebblies** [comment]
Anyone?  This is blocking my work on cleaning up varargs, which is blocking ddmd.  I recommend viewing with [whitespace ignored](https://github.com/D-Programming-Language/druntime/pull/1102/files?w=1).

File: src/rt/lifetime.d [hunk 1]
```diff
@@ -1116,7 +1086,44 @@ extern (C) void[] _d_newarraymiT(const TypeInfo ti, size_t ndims, ...)
...
+    {
+        return _d_newarrayOpT!(_d_newarrayT)(ti, dims);
+    }
+}
+
+
+/**
+ *
+ */
+extern (C) void[] _d_newarraymiTX(const TypeInfo ti, size_t[] dims)
```
Comment (MartinNowak) [inline src/rt/lifetime.d]: How about typesafe variadic `size_t[] dims...` instead? Because that's how you're effectively calling these functions

Comment (yebblies) [inline src/rt/lifetime.d]: There's no point as these never get called directly from D code.  The code to stack-allocate the arrays has to be created manually in the glue layer anyway.

Comment (MartinNowak) [inline src/rt/lifetime.d]: Sure, it doesn't change anything currently, but it reflects what's done.

Comment (yebblies) [inline src/rt/lifetime.d]: The way I see it, making the function typesafe variadic would be misleading, as that type of function doesn't actually exist at dmd's elem-level.

When you say currently, do you mean you intend to generate these calls at a higher level inside dmd?

Comment (MartinNowak) [inline src/rt/lifetime.d]: No, I had escape analysis in mind.
I don't see much benefit in moving these calls to a higher level. Sure we could replace the TypeInfo stuff with templated functions, but that hardly improves anything, except for attribute inference of postblit calls.

Comment (yebblies) [inline src/rt/lifetime.d]: Oh ok, that makes more sense.
The whole AA mess has made me wary of higher-level lowerings, although it's the only way to get dmd's inliner to look at runtime calls.  I think in the long run we'll drop the typeinfo interface, but I'm not in a hurry.

**Comment by ibuclaw** [comment]
I should have looked here earlier.  FYI - GDC passes the length and ptr as separate parameters, not that it makes a difference.

this https://github.com/D-Programming-GDC/GDC/blob/e6130a9c04e308ec5e5e238d6355061f682b7b34/libphobos/libdruntime/rt/lifetime.d#L978

**Comment by yebblies** [comment]
Hahaha, is it just me or is your version ABI-compatible with this version on every dmd platform except win64?

**Comment by ibuclaw** [comment]
Why would you think that? :)

**Comment by yebblies** [comment]
Fond memories of `"%.*s"`.

**Pull request merged**